### PR TITLE
Use ruff instead of black for autoformatting

### PR DESCRIPTION
**Pull request by zaneselvans**
# PR Overview

Ruff now provides [very fast code formatting](https://astral.sh/blog/the-ruff-formatter) in addition to linting + autofixes, with "99.9%" compatibility with `black`, intended to be used as a drop-in replacement.

This PR updates our environment configuration and pre-commit hooks to use `ruff` as the autoformatter instead of `black` and applies a few minor resulting changes to the repo.

# PR Checklist

- [ ] Merge the most recent version of the branch you are merging into (probably `dev`).
- [ ] All CI checks are passing. [Run tests locally to debug failures](https://catalystcoop-pudl.readthedocs.io/en/latest/dev/testing.html#running-tests-with-tox)
- [ ] Make sure you've included good docstrings.
- [ ] For major data coverage & analysis changes, [run data validation tests](https://catalystcoop-pudl.readthedocs.io/en/latest/dev/testing.html#data-validation)
- [ ] Include unit tests for new functions and classes.
- [ ] Defensive data quality/sanity checks in analyses & data processing functions.
- [ ] Update the [release notes](https://catalystcoop-pudl.readthedocs.io/en/latest/release_notes.html) and reference reference the PR and related issues.
- [ ] Do your own explanatory review of the PR to help the reviewer understand what's going on and identify issues preemptively.

### CODE REVIEW

**Comment by zaneselvans** [comment]
Hmm, the test failures seem to be unrelated to the PR, and due to an issue with the micromamba setup action, which is complaining about not knowing which kind of hash to use, md5 or sha256...

**Comment by zaneselvans** [comment]
I've opened an issue in the `conda-lock` repo: https://github.com/conda/conda-lock/issues/556

File: environments/conda-lock.yml [hunk 1]
```diff
@@ -1819,60 +1819,6 @@ package:
       sha256: 52d3e6bcd442537e22699cd227d8fdcfd54b708eeb8ee5b4c671a6a9b9cd74da
     category: main
     optional: false
-  - name: black
```
Comment (jdangerx) [inline environments/conda-lock.yml:1822]: One fewer dependency! It does make me feel a little strange to leave `black` behind, I had bought in so hard to "just one formatter, deal with it, screw your style guide." But happy to have a new benevolent style dictator in my life :)

File: pyproject.toml [hunk 2]
```diff
@@ -212,6 +211,10 @@ line-length = 88
 # Don't automatically concatenate strings -- sometimes we forget a comma!
 unfixable = ["ISC"]
 
+[tool.ruff.format]
+quote-style = "double"
+exclude = ["migrations/versions/*"]
```
Comment (jdangerx) [inline pyproject.toml:216]: YES thank you for excluding the generated files! This always mildly bugged me and then I would forget to fix it 🙄

**Review by jdangerx — APPROVED** [review]
friendship ended with BLACK, now RUFF is my best friend

**Pull request merged**